build.gradle.kts

```diff
@@ -71,7 +71,10 @@ subprojects {
   }
 
   tasks.withType<KotlinCompile> {
-    kotlinOptions.jvmTarget = JavaVersion.VERSION_1_8.toString()
+    kotlinOptions {
+      jvmTarget = JavaVersion.VERSION_1_8.toString()
+      freeCompilerArgs += "-Xopt-in=kotlin.RequiresOptIn"
```

Comment:
what is the implication on the user side, will it be required to include this compiler arg on the library consumer side as well?

Reply:
As far as understand, this makes it an implementation detail. Users shouldn't require anything. We can test that with snapshots after merging this one. If it is not the case, it is easy to remove this. The experimental map/set builder API is just nicer.

Reply:
I just want to make sure that we don't introduce extra requirements for the users to include this compilation flag on their end whenever they want to use `dump()`. 

https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-experimental-stdlib-api/
> Beware using the annotated API especially if you're developing a library, since your library might become binary incompatible with the future versions of the standard library.

It can cause issues in the future if they decide to change API.

Reply:
Tbh, this is an inline function and it is not something we expose to consumers.In this particular case, it should be fine.